parser/src/ast/boundary_constraints.rs

```diff
@@ -77,3 +71,28 @@ pub enum BoundaryExpr {
     Mul(Box<BoundaryExpr>, Box<BoundaryExpr>),
     Exp(Box<BoundaryExpr>, u64),
 }
+
+#[derive(Debug, PartialEq)]
+pub struct BoundaryVariable {
+    name: Identifier,
+    value: BoundaryVariableType,
+}
+
+impl BoundaryVariable {
+    pub fn new(name: Identifier, value: BoundaryVariableType) -> Self {
+        Self { name, value }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub enum BoundaryVariableType {
+    Scalar(BoundaryExpr),
+    Vector(Vec<BoundaryExpr>),
+    Matrix(Vec<Vec<BoundaryExpr>>),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum BoundaryStmt {
+    Constraint(BoundaryConstraint),
+    Variable(BoundaryVariable),
+}
```

Comment:
nit: I would probably put this towards the top of the file as this is the top level struct.